### PR TITLE
DEV-821 Add option to specify CMEK

### PR DESCRIPTION
**Pull request by pauldwolfe**
Passed in as an argument, when present all runtime buckets will set the
kms property on creation (note will not change old buckets already using
Google keys).

Also necessary to increase storage timeouts and use the .getResult() to
ensure the the entire copy operation completes.

### CODE REVIEW

File: cluster/src/test/java/com/hartwig/pipeline/storage/RuntimeBucketTest.java [hunk 1]
```diff
@@ -133,6 +133,19 @@ public void namespacesListOperationPrefix() {
         assertThat(listOptionCaptor.getValue()).isEqualTo(Storage.BlobListOption.prefix("test/prefix"));
     }
 
+    @Test
+    public void usesCmekWhenProvided() {
+        RuntimeBucket.from(storage, NAMESPACE, referenceRunMetadata(), Arguments.testDefaultsBuilder().cmek("key").build());
+        assertThat(bucketInfo.getValue().getDefaultKmsKeyName()).isEqualTo(
+                "projects/hmf-pipeline-development/locations/europe-west4/keyRings/hmf-pipeline-development/cryptoKeys/key");
```
Comment (nedleitch) [inline cluster/src/test/java/com/hartwig/pipeline/storage/RuntimeBucketTest.java]: Do we want to parameterise this? Makes the test somewhat brittle if we don't.

File: cluster/src/main/java/com/hartwig/pipeline/storage/StorageProvider.java [hunk 2]
```diff
@@ -16,7 +17,15 @@ private StorageProvider(final Arguments arguments, final GoogleCredentials crede
     }
 
     public Storage get() {
-        return StorageOptions.newBuilder().setCredentials(credentials).setProjectId(arguments.project()).build().getService();
+        StorageOptions.Builder builder = StorageOptions.newBuilder();
+        return builder.setCredentials(credentials)
+                .setProjectId(arguments.project())
+                .setTransportOptions(HttpTransportOptions.newBuilder()
+                        .setConnectTimeout(Integer.MAX_VALUE)
+                        .setReadTimeout(Integer.MAX_VALUE)
```
Comment (nedleitch) [inline cluster/src/main/java/com/hartwig/pipeline/storage/StorageProvider.java]: I think you can pass "0" here if you want an infinite timeout.

[reference](https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-core-http/src/main/java/com/google/cloud/http/HttpTransportOptions.java)

Comment (pauldwolfe) [reply cluster/src/main/java/com/hartwig/pipeline/storage/StorageProvider.java]: Good idea

File: cluster/src/test/java/com/hartwig/pipeline/storage/RuntimeBucketTest.java (hunk 1 above)
Comment (pauldwolfe) [reply cluster/src/test/java/com/hartwig/pipeline/storage/RuntimeBucketTest.java]: Struggle a bit with this, as if I parameterize it, I basically end up rewriting the code I wrote in the RuntimeBucket itself. But maybe that's still better than fixing a million littles tests if the Arguments.testDefaults() change. I'll implement your suggestion

**Pull request merged**